js/main.js

```diff
@@ -0,0 +1,108 @@
+
+  }
+  /*   console.log('villagerData:', villagerData); */
+
+});
+xhrVillagers.send();
+
+var $newVillagerSearched = document.querySelector('.navbar-form');
+$newVillagerSearched.addEventListener('submit', saveForm);
+function saveForm(event) {
+  event.preventDefault();
+
+  var formValues = {};
+  formValues.name = $newVillagerSearched.elements.name.value;
+
+  var villagerNamesList = Object.keys(villagerData);
+  for (var i = 0; i < villagerNamesList.length; i++) {
+    if (formValues.name === villagerNamesList[i]) {
+
+      formValues.entryId = data.nextEntryId;
+      formValues.hobby = villagerData[villagerNamesList[i]].hobby;
+      formValues.personality = villagerData[villagerNamesList[i]].personality;
+      formValues.icon = villagerData[villagerNamesList[i]].icon;
+      formValues.image = villagerData[villagerNamesList[i]].image;
+      formValues.birthdayString = villagerData[villagerNamesList[i]].birthdayString;
+
+      formValues.saying = villagerData[villagerNamesList[i]].saying;
+
+      data.nextEntryId++;
+      data.entries.unshift(formValues);
+      document.querySelector('.navbar-form').reset();
+      renderVillagerSearch();
+      break;
+    } else if (formValues.name !== villagerNamesList[i]) {
+      renderNotFound();
+    }
+  }
+
+}
+
+var $searchImage = document.querySelector('.search-image');
+var $icon = document.querySelector('.icon');
+var $name = document.querySelector('.villager-name');
+var $birthday = document.querySelector('.birthday');
+var $hobbyText = document.querySelector('.hobby-text');
+var $personalityText = document.querySelector('.personality-text');
+var $sayingText = document.querySelector('.saying-text');
+
+function renderVillagerSearch(entry) {
+
+  $searchImage.setAttribute('src', data.entries[0].image);
+  $icon.setAttribute('src', data.entries[0].icon);
+  $name.textContent = data.entries[0].name;
+  $name.classList.remove('text-red');
+  $birthday.textContent = data.entries[0].birthdayString;
+  $hobbyText.textContent = data.entries[0].hobby;
+  $personalityText.textContent = data.entries[0].personality;
+  $sayingText.textContent = data.entries[0].saying;
+  $heart.classList.remove('hidden');
+  $heart.className = 'heart-like fa-regular fa-heart centered house-font-size';
+}
+
+function renderNotFound(entry) {
+  $searchImage.setAttribute('src', '/images/sad.png');
+  $icon.setAttribute('src', '/images/thought.png');
+  $name.textContent = 'No villager was found!';
+  $name.classList.add('text-red');
+  $birthday.textContent = '';
+  $hobbyText.textContent = '';
+  $personalityText.textContent = '';
+  $sayingText.textContent = '';
+  $heart.classList.add('hidden');
+  $textHeart.className = 'text-heart transparent';
+}
+var clicks = 0;
+var $heart = document.querySelector('.heart-like');
+var $textHeart = document.querySelector('.text-heart');
+$heart.addEventListener('click', heartClicked);
+function heartClicked(event) {
+  clicks++;
+  if (clicks % 2 !== 0) {
```

Comment:
why the click count here? You aren't using the count for anything. Instead, you should be making this heart filled or not filled based on looking at your list of saved villagers and seeing if the villager you are looking at is in that list.